access/resource_ip_access_list.go

```diff
@@ -105,30 +31,47 @@ func ResourceIPAccessList() *schema.Resource {
 	return common.Resource{
 		Schema: s,
 		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			var iacl createIPAccessListRequest
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var iacl settings.CreateIpAccessList
 			common.DataToStructPointer(d, s, &iacl)
-			status, err := NewIPAccessListsAPI(ctx, c).Create(iacl)
+			status, err := w.IpAccessLists.Create(ctx, iacl)
 			if err != nil {
 				return err
 			}
-			d.SetId(status.ListID)
+			d.SetId(status.IpAccessList.ListId)
 			return nil
 		},
 		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			status, err := NewIPAccessListsAPI(ctx, c).Read(d.Id())
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			status, err := w.IpAccessLists.GetByIpAccessListId(ctx, d.Id())
 			if err != nil {
 				return err
 			}
 			common.StructToData(status, s, d)
 			return nil
 		},
 		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			var iacl ipAccessListUpdateRequest
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var iacl settings.UpdateIpAccessList
 			common.DataToStructPointer(d, s, &iacl)
-			return NewIPAccessListsAPI(ctx, c).Update(d.Id(), iacl)
+			iacl.IpAccessListId = d.Id()
```

Comment:
We need to do this because Update call doesn't take id so we have to pass it through the resource data.
Any reason why we don't pass id through the Update method call?